docs/03.API-EXAMPLE.md

```diff
@@ -20,39 +54,211 @@ main (void)
 }
 ```
 
-The application will return with zero exit code.
+To compile it one can use the following command:
+
+```sh
+$ gcc api-example-1.c -o api-example-1 $(pkg-config --cflags --libs libjerry-core libjerry-port-default libjerry-libm)
+```
+
+If everything is correct the application returns with a zero exit code:
 
-## Step 2. Split engine initialization and script execution
+```
+$ ./api-example-1
+$ echo $?
+```
 
-Here we perform the same actions, as `jerry_run_simple`, while splitting into several steps:
+## Example 2. Split engine initialization and script execution.
 
-- engine initialization
-- script code setup
-- script execution
-- engine cleanup
+In this example the engine is initialized directly with the `jerry_init` method
+and cleaned up with the `jerry_cleanup` method. The example JavaScript code
+is directly parsed and executed via the `jerry_eval` method. Each `jerry_value_t`
+returned by the API methods is freed with the `jerry_release_value` method.
 
+To make sure that the code parsing and execution was ok, the `jerry_value_is_error`
+method is used to check for any errors.
+
+Use the following code for the `api-example-2.c` file:
 
 [doctest]: # ()
 
 ```c
 #include "jerryscript.h"
-#include "jerryscript-ext/handler.h"
 
 int
 main (void)
 {
-  const jerry_char_t script[] = "print ('Hello, World!');";
+  const jerry_char_t script[] = "var str = 'Hello, World!';";
+  const jerry_length_t script_size = sizeof (script) - 1;
+  /* Note: sizeof only usable as the compiler knows the static character array's size! */
```

Comment:
"sizeof can be used here only because the compiler knows the static character arrays's size. If this is not the case, strlen should be used instead."?